### PR TITLE
Extracting druid minting from ObjectCreator

### LINKED ISSUE #3511: Extract minting pid from ObjectCreator
Minting the pid should be moved to `CocinaObjectStore.create`. It is going to need to be minted regardless of whether persisting to Fedora or PG.

### PR DESCRIPTION
**Pull request by lwrubel**
## Why was this change made? 🤔
Resolves #3511 to move druid minting from ObjectCreator to CocinaObjectStore.

## How was this change tested? 🤨
Specs pass (made two changes, however).

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/services/cocina/object_creator.rb [hunk 1]
```diff
@@ -4,13 +4,17 @@ module Cocina
...
+    def self.create(cocina_object, druid:, event_factory: EventFactory, persister: ActiveFedoraPersister, assign_doi: false, cocina_object_store: CocinaObjectStore.new)
+      _fedora_object, cocina_object = new(cocina_object_store: cocina_object_store).create(cocina_object, druid: druid, event_factory: event_factory, persister: persister,
+                                                                                                          assign_doi: assign_doi)
       cocina_object
     end
+    # rubocop:enable Metrics/ParameterLists
 
     def self.trial_create(cocina_object, notifier:, cocina_object_store:)
-      new(cocina_object_store: cocina_object_store).create(cocina_object, event_factory: nil, persister: nil, trial: true, notifier: notifier)
+      druid = cocina_object.externalIdentifier
```
Comment (justinlittman) [inline app/services/cocina/object_creator.rb:16]: No real reason to assign this to a variable.

File: app/services/cocina/object_creator.rb [hunk 2]
```diff
@@ -20,10 +24,10 @@ def initialize(cocina_object_store: CocinaObjectStore.new)
     # @param [Cocina::Models::RequestDRO,Cocina::Models::RequestCollection,Cocina::Models::RequestAdminPolicy] cocina_object
     # @raises SymphonyReader::ResponseError if symphony connection failed
     # rubocop:disable Metrics/ParameterLists
-    def create(cocina_object, event_factory:, persister:, trial: false, notifier: nil, assign_doi: false)
+    def create(cocina_object, druid:, event_factory:, persister:, trial: false, notifier: nil, assign_doi: false)
```
Comment (justinlittman) [inline app/services/cocina/object_creator.rb:27]: Add param to docs?

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

File: app/services/cocina/object_creator.rb (hunk 1 above)
Comment (lwrubel) [reply app/services/cocina/object_creator.rb:16]: Ah, I see what you mean. Fixing.

**Pull request merged**